UM/Settings/Interfaces.py

```diff
@@ -1,11 +1,40 @@
 # Copyright (c) 2016 Ultimaker B.V.
 # Uranium is released under the terms of the AGPLv3 or higher.
 
+from collections import deque
 from typing import List, Dict, Any, Optional
 
 import UM.Decorators
 from UM.Signal import Signal
 
+##  Context for evaluating a property value
+#   It contains:
+#     1. a stack of containers during the evaluation in the function call stack fashion
+#     2. a context dictionary which contains all the current context
+#
+class PropertyEvaluationContext:
```

Comment:
Why is this in `Interfaces.py`? There is nothing interface about this.

Reply:
Fixed in 068f7f6e05f5422971036b5fabe7a74001901ba5